### PR TITLE
chore(tests): investigate karma tests don't run locally

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally ~and any changes were pushed~

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): N/A

## What is the current behavior?

Running `npm run test.prod` would fail locally. The reason for which was the subcommand `test.karma.prod` would fail.  This had a few implications:
- we couldn't run the our entire test suite locally
- we couldn't _release_ new versions of Stencil

## What is the new behavior?

- `npm ci` is now run as a part of the karma test scripts when running locally
  - [we do this in CI today, which is why is passed there](https://github.com/ionic-team/stencil/blob/ccf4c57a619598b6f13cf485747390ce7ab065f2/.github/workflows/main.yml#L63-L71)
- invokes the tests
  - this was [removed last year](https://github.com/ionic-team/stencil/commit/8686b4cab5c881aa01a783b30ffc53931e8a0dd3)
  - one test has been intentionally skipped, as its broken. I would rather 87 passing tests and one skipped one (with an item in the backlog to fix it) rather than none running if that's our only alternative 😉 .
  
## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

I was able to run `npm run test.karma.prod` locally.

### CODE REVIEW

**Comment by rwaskiewicz** [comment]
@johnjenkins thanks!

File: test/karma/test-app/slot-fallback/karma.spec.ts
Comment (rwaskiewicz) [inline test/karma/test-app/slot-fallback/karma.spec.ts:5]: FWIW I normally prefer the `.skip` syntax here, but we're using Jasmine here, which doesn't have that in it's API

File: package.json
Comment (ltm) [inline package.json:46]: Should we be running `test.karma.prod` in CI as well (i.e. change this to `npm run test.karma.prod`)?
https://github.com/ionic-team/stencil/blob/ccf4c57a619598b6f13cf485747390ce7ab065f2/.github/workflows/main.yml#L71

Comment (rwaskiewicz) [reply package.json:46]: 👍 done in https://github.com/ionic-team/stencil/pull/2956/commits/4bc1011d76b038f51fad1c9ab6ae4effb2e06a72

File: test/karma/package.json
Comment (rwaskiewicz) [inline test/karma/package.json:9]: This was originally under a script named `tsc` - at first read I thought it was invoking the TS compiler directly, so I renamed it for clarity. This appears to only be directly used for the test-app, hence the name.  I did consider moving the `tsconfig.json` itself down to the `test-app/`, but decided against it as it's referenced in other places and felt out of scope of this user story

**Comment by splitinfinities** [comment]
Could you squash each commits down into one? Afterward, looks great! 👍

**Comment by rwaskiewicz** [comment]
@splitinfinities for sure, I normally do that at merge time, that way I'm not rewriting the git history until everyone's approved (that way if one person approves and the other asks for changes, we're all talking about the same lineage of commits)

**Review by ltm — APPROVED** [review]

**Review by splitinfinities — APPROVED** [review]
Looks good!

**Pull request merged**
